src/main/haskell/kore/src/Kore/Builtin/List.hs

```diff
@@ -457,17 +457,17 @@ unify
         -> Seq p
         -> p
         -> m (expanded, proof)
-    unifyFramedRight resultSort list1 symConcat prefix2 frame2
+    unifyFramedRight resultSort list1 _symConcat prefix2 frame2
       | Seq.length prefix2 > Seq.length list1 =
         return (ExpandedPattern.bottom, SimplificationProof)
       | otherwise =
         do
             (prefixUnified, _) <- unifyConcrete resultSort prefix1 prefix2
             (suffixUnified, _) <- simplifyChild frame2 listSuffix1
-            let result =
-                    (<$>)
-                        (App_ symConcat)
-                        (propagatePredicates [prefixUnified, suffixUnified])
+            let result = give symbolOrAliasSorts $
+                    pure (DV_ resultSort (BuiltinDomainList list1))
```

Comment:
Could we pass in the original `DV_` pattern so that it is shared instead of reconstructing it?